packages/compass-aggregations/src/modules/pipeline-builder/pipeline-parser/pipeline-parser.spec.ts

```diff
@@ -497,4 +497,33 @@ describe('PipelineParser', function () {
       });
     });
   });
+
+  it('parses correctly when a pipeline with disabled stages is added another stage', function () {
+
+    const pipeline = `[
+      // { $match: { name: /berlin/i } },
+      // { $unwind: "users" },
+    ]`;
```

Comment:
Let's maybe add a few other cases just in case: maybe some non-stage leading / trailing comments

Reply:
I found another issue while adding tests. An empty pipeline such as:

```
[
	// {$limit: 10},
	// some comment
]
```

after switching to Builder Mode and then back to Text mode, the comment is lost in the process. 
The initial fix in this PR fixed it, but it pushed the comment to the top. The [ccffb33](https://github.com/mongodb-js/compass/pull/3819/commits/ccffb3351efb6c9a687470cf84b5630b3951b3ab) fixes it as well.